code/src/test/kotlin/com/expediagroup/sdk/core/logging/masking/MaskingPatternBuilderTest.kt

```diff
@@ -0,0 +1,227 @@
+package com.expediagroup.sdk.core.logging.masking
+
+import com.expediagroup.sdk.core.logging.common.Constant.OMITTED
+import org.junit.jupiter.api.assertAll
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class MaskingPatternBuilderTest {
+    @Nested
+    inner class PatternGenerationTest {
+        @Test
+        fun `generated masking patterns are empty if no fields are passed`() {
+            assertEquals(0, MaskingPatternBuilder().build().size)
+        }
+
+        @Test
+        fun `generated masking patterns size is one when only global fields are passed regardless of their count`() {
+            val globalFields = setOf("globalField1", "globalField2")
+            val patterns = MaskingPatternBuilder().globalFields(*globalFields.toTypedArray()).build()
+
+            assertEquals(1, patterns.size)
+        }
+
+        @Test
+        fun `passed path fields are truncated to the last two elements`() {
```

Comment:
Please document that this test should fail once support for more elements is added.